### PR TITLE
[J#PEAK-1246] Silence RestrictedFilter sniff, handled by is_array

### PR DESCRIPTION
**Pull request by a6software**
VIPBot warning:

⚠️ Warning: Please use an appropriate filter to sanitize, as "FILTER_DEFAULT" does no filtering, see: http://php.net/manual/en/filter.filters.sanitize.php (WordPressVIPMinimum.Security.PHPFilterFunctions.RestrictedFilter).

Suppressed by this change. 

Handled on line 339 by `is_array` check.

### CODE REVIEW

File: includes/class-gdpr.php [hunk 1]
```diff
@@ -331,9 +331,11 @@ public static function similar_in_array( $needle, $haystack ) {
 	public static function save_user_consent_on_registration( $user_id ) {
 		GDPR_Audit_Log::log( $user_id, esc_html__( 'User registered to the site.', 'gdpr' ) );
 
+		// phpcs:disable WordPressVIPMinimum.Security.PHPFilterFunctions.RestrictedFilter
 		// phpcs:disable WordPressVIPMinimum.VIP.PHPFilterFunctions.RestrictedFilter
 		$user_consents = filter_input( INPUT_POST, 'user_consents', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY );
 		// phpcs:enable WordPressVIPMinimum.VIP.PHPFilterFunctions.RestrictedFilter
+		// phpcs:enable WordPressVIPMinimum.Security.PHPFilterFunctions.RestrictedFilter
```
Comment (jenkoian) [inline includes/class-gdpr.php]: Can we change `FILTER_DEFAULT` to `FILTER_SANITIZE_STRING` instead of just disabling?

Comment (a6software) [reply includes/class-gdpr.php]: We don't seem to use it, it always seems to be `null`. What's concerning me from simply removing it is I can't figure out how it is supposed to be used, and of course, there are no unit tests so figuring out the way it's supposed to work is a crapshoot. 

In short:

`GDPR_Audit_Log::log( $user_id, esc_html__( 'User registered to the site.', 'gdpr' ) );`

is used.

But as the `$user_contents` outcome is always(?) `null`, then the conditional is always `false`, so all we have here is a VIP warning with absolutely no benefit to production code.

Comment (a6software) [reply includes/class-gdpr.php]: ok figured it out, it's for if we were using custom consents on registration. We aren't, I've removed the entire conditional.

Comment (jenkoian) [reply includes/class-gdpr.php]: we aren't but if it's supported we should leave the plugin be fairly agnostic?

Comment (a6software) [reply includes/class-gdpr.php]: ok, no problem

**Review by jenkoian — APPROVED** [review]

**Pull request merged**